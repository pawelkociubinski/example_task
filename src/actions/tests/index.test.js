import * as ActionCreator from '../index';

describe('Action Creators', () => {
  it('searchTerm', () => {
    expect(ActionCreator.searchTerm('react')).toMatchSnapshot();
  });

  it('sortByColumn', () => {
    expect(ActionCreator.sortByColumn('id', 'asc')).toMatchSnapshot();
  });

  it('changePage', () => {
    expect(ActionCreator.changePage(1)).toMatchSnapshot();
  });

  it('previousPage', () => {
    expect(ActionCreator.previousPage()).toMatchSnapshot();
  });

  it('nextPage', () => {
    expect(ActionCreator.nextPage()).toMatchSnapshot();
  });

  it('getGithubToken', () => {
    expect(ActionCreator.getGithubToken('github_code')).toMatchSnapshot();
  });

  it('sortItemsPerPage', () => {
    expect(ActionCreator.sortItemsPerPage(1)).toMatchSnapshot();
  });
});
