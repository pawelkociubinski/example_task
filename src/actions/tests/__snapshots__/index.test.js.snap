// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action Creators changePage 1`] = `
Object {
  "payload": Object {
    "page": 1,
  },
  "type": "CHANGE_PAGE",
}
`;

exports[`Action Creators getGithubToken 1`] = `
Object {
  "payload": Object {
    "code": "github_code",
  },
  "type": "GITHUB_TOKEN_REQUESTED",
}
`;

exports[`Action Creators nextPage 1`] = `
Object {
  "type": "NEXT_PAGE",
}
`;

exports[`Action Creators previousPage 1`] = `
Object {
  "type": "PREVIOUS_PAGE",
}
`;

exports[`Action Creators searchTerm 1`] = `
Object {
  "payload": Object {
    "searchedTerm": "react",
  },
  "type": "SEARCH_TERM_REQUESTED",
}
`;

exports[`Action Creators sortByColumn 1`] = `
Object {
  "payload": Object {
    "orderBy": "asc",
    "sortByProperty": "id",
  },
  "type": "SORT_COLUMN_BY_ORDER",
}
`;

exports[`Action Creators sortItemsPerPage 1`] = `
Object {
  "payload": Object {
    "itemsPerPage": 1,
  },
  "type": "SORT_ITEMS_PER_PAGE",
}
`;
